source/extensions/quic_listeners/quiche/platform/envoy_quic_clock.cc

```diff
@@ -0,0 +1,22 @@
+#include "extensions/quic_listeners/quiche/platform/envoy_quic_clock.h"
+
+namespace Envoy {
+namespace Quic {
+
+quic::QuicTime EnvoyQuicClock::ApproximateNow() const {
+  // This might be expensive as Dispatcher doesn't store approximate time_point.
+  return Now();
```

Comment:
worst case we can have a thread local alarm which latches time once per dispatcher loop.

Reply:
LibeventScheduler::onPrepare() is called each libevent loop. But there are some plumbing to pass the timestamp get from onPrepare() to TimeSystem. But it's doable, and can be an future improvement.